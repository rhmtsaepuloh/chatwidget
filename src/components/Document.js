/* eslint-disable react/jsx-no-target-blank */
/* eslint-disable jsx-a11y/img-redundant-alt */
import React from "react";
import { FaFilePdf, FaFileExcel, FaFileAlt } from "react-icons/fa";
import isEmpty from "lodash/isEmpty";
import { MdDoneAll, MdInsertDriveFile } from "react-icons/md";
import RenderDocument from "./RenderDocument";

const Document = ({
  extension,
  color,
  timestamp,
  text,
  size,
  caption,
  downloadFile,
}) => {
  const splitExtension = isEmpty(extension) ? null : extension.split(".").pop();

  const SwitchDocument = () => {
    switch (splitExtension) {
      case "pdf":
        return (
          <RenderDocument
            icon={<FaFilePdf style={{ color }} />}
            text={text}
            extension={extension}
            splitExtension={splitExtension}
            size={size}
            timestamp={timestamp}
            downloadFile={downloadFile}
            caption={caption}
          />
        );
      case "doc":
      case "docx":
        return (
          <RenderDocument
            icon={<MdInsertDriveFile style={{ color }} />}
            text={text}
            extension={extension}
            splitExtension={splitExtension}
            size={size}
            timestamp={timestamp}
            downloadFile={downloadFile}
            caption={caption}
          />
        );
      case "txt":
        return (
          <RenderDocument
            icon={<FaFileAlt style={{ color }} />}
            text={text}
            extension={extension}
            splitExtension={splitExtension}
            size={size}
            timestamp={timestamp}
            downloadFile={downloadFile}
            caption={caption}
          />
        );
      case "xls":
      case "xlsx":
        return (
          <RenderDocument
            icon={<FaFileExcel style={{ color }} />}
            text={text}
            extension={extension}
            splitExtension={splitExtension}
            size={size}
            timestamp={timestamp}
            downloadFile={downloadFile}
            caption={caption}
          />
        );
      default:
        return (
          <React.Fragment>
            <img
              src={extension}
              alt='image'
              className='h-56 w-full mb-2 rounded-md'
            />
            <p className='whitespace-pre-wrap break-words'>{caption}</p>
            <div className='file-timestamp flex items-center justify-end'>
              <p className='text-sm' style={{ color: "#979797" }}>
                {timestamp}
              </p>
              <MdDoneAll className='ml-1' />
            </div>
          </React.Fragment>
        );
    }
  };
  return SwitchDocument();
};

export default Document;
