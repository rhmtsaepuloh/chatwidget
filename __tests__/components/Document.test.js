/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import Document from "../../src/components/Document";

const props = {
  color: "#000",
  timestamp: "10:21",
  text: "kirim file",
  size: 781076,
  caption: "tambahkan caption",
};

describe("render correctly document components", () => {
  test("renders if not exist extension", () => {
    const extension = null;
    const wrapper = shallow(<Document {...props} extension={extension} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file image", () => {
    const extension = "file.png";
    const wrapper = shallow(<Document {...props} extension={extension} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file pdf", () => {
    const extension = "file.pdf";
    const wrapper = shallow(<Document {...props} extension={extension} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file xls", () => {
    const extension = "file.xls";
    const wrapper = shallow(<Document {...props} extension={extension} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file xlsx", () => {
    const extension = "file.xlsx";
    const wrapper = shallow(<Document {...props} extension={extension} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file doc", () => {
    const extension = "file.doc";
    const wrapper = shallow(<Document {...props} extension={extension} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file docx", () => {
    const extension = "file.docx";
    const wrapper = shallow(<Document {...props} extension={extension} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file txt", () => {
    const extension = "file.txt";
    const wrapper = shallow(<Document {...props} extension={extension} />);
    expect(wrapper).toMatchSnapshot();
  });
});
