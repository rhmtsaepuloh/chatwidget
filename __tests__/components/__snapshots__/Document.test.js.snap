// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly document components renders file doc 1`] = `ShallowWrapper {}`;

exports[`render correctly document components renders file docx 1`] = `ShallowWrapper {}`;

exports[`render correctly document components renders file image 1`] = `ShallowWrapper {}`;

exports[`render correctly document components renders file pdf 1`] = `ShallowWrapper {}`;

exports[`render correctly document components renders file txt 1`] = `ShallowWrapper {}`;

exports[`render correctly document components renders file xls 1`] = `ShallowWrapper {}`;

exports[`render correctly document components renders file xlsx 1`] = `ShallowWrapper {}`;

exports[`render correctly document components renders if not exist extension 1`] = `ShallowWrapper {}`;
